### PR TITLE
Adds a sprint buffer - 35 tiles, regen 2 tiles/second (always active), 0.3 stamcost per tile if empty (as old) directly applied to staminaloss rather than stamina action buffer (unlike old)

### PR DESCRIPTION
**Pull request by silicons**
Title
Values up to debate.

### CODE REVIEW

File: modular_citadel/code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/mob/living/carbon/Life()
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/living/carbon/life.dm:1]: Shouldn't you also be calling `. = ..()` here?

File: modular_citadel/code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -8,6 +8,14 @@
 	var/intentionalresting = FALSE
 	var/attemptingcrawl = FALSE
 
+	//Sprint buffer---
+	var/sprint_buffer = 35					//Tiles
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/living/living.dm]: Suggesting you to turn this into a define.

File: modular_citadel/code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -8,6 +8,14 @@
...
+	var/sprint_buffer_max
+	var/sprint_buffer_regen_ds = 0.2		//Tiles per world.time decisecond
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/living/living.dm]: Suggesting you to turn this value into a define.

File: modular_citadel/code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -8,6 +8,14 @@
...
+	var/sprint_buffer_regen_last = 0		//last world.time this was regen'd for math.
+	var/sprint_stamina_cost = 0.3			//stamina loss per tile while insufficient sprint buffer.
```
Comment (Ghommie) [inline modular_citadel/code/modules/mob/living/living.dm]: And this one too.

**Comment by Poojawa** [comment]
I still dislike the band-aid like treatment to a core issue that this feels like, but it won't hurt to try it out at least.

File: modular_citadel/code/modules/mob/living/living.dm (hunk 3 above)
Comment (silicons) [reply modular_citadel/code/modules/mob/living/living.dm]: it's variable for a reason same below

**Comment by Poojawa** [comment]
So begins the power creep :^)

**Comment by peeesh** [comment]
> So begins the power creep :^)

no

**Comment by Poojawa** [comment]
> > So begins the power creep :^)
> 
> no

Woosh.

**Comment by Ghommie** [comment]
Currently the sprinter buffer ui icon doesn't take in account if the widescreen is NOT enabled and get overlayered by other ui buttons. Please fix that, thanks.

**Comment by silicons** [comment]
oh yeah it's fixed @deathride58

**Review by deathride58 — DISMISSED** [review]

**Review by deathride58 — APPROVED** [review]

**Pull request merged**